src/test/java/org/minions/devfund/marylin/Squarelotron.java

```diff
@@ -0,0 +1,267 @@
+     */
+    void printSquarelotron() {
+        for (int i = 0; i < squarelotron.length; i++) {
+            for (int j = 0; j < squarelotron.length; j++) {
+                System.out.print(" " + squarelotron[i][j] + " ");
+            }
+            System.out.println();
+        }
+    }
+
+    /**
+     * upsideDownFlipCompleteMatrix without check rings.
+     *
+     * @param matrix array.
+     * @return the matrix that is upside down flip completely.
+     */
+    int[][] upsideDownFlipCompleteMatrix(final int[][] matrix) {
+        int[][] resultGrid = duplicateGrid(matrix);
+        int size = matrix.length;
+        int temp = 0;
+        for (int i = size - 1; i >= size / 2; i--) {
+            for (int j = 0; j < size; j++) {
+                resultGrid[i][j] = matrix[temp][j];
+                resultGrid[temp][j] = matrix[i][j];
+            }
+            temp++;
+        }
+        return resultGrid;
+    }
+
+    /**
+     * upsideDownFlipInternalMatrix distinc to ring #1.
+     *
+     * @param matrix to be upside down flip the internal matrix.
+     * @return the matrix result-
+     */
+    int[][] upsideDownFlipInternalMatrix(final int[][] matrix) {
+        int[][] resultGrid = duplicateGrid(matrix);
+        int size = matrix.length;
+        int temp = 1;
+        for (int i = size - 2; i >= size / 2; i--) {
+            for (int j = 1; j < size - 1; j++) {
+                resultGrid[i][j] = matrix[temp][j];
+                resultGrid[temp][j] = matrix[i][j];
+            }
+            temp++;
+        }
+        return resultGrid;
+    }
+
+    /**
+     * Upside Down Flip.
+     *
+     * @param ring number.
+     * @return an instance of Squarelotron.
+     */
+    Squarelotron upsideDownFlip(int ring) {
+        int[][] resultGrid = duplicateGrid(this.squarelotron);
+        int size = resultGrid.length;
+
+        //Returns the same squarelotron if the ring number is equal to the number of rings of a matrix n x n.
+        if (size % 2 == 1 && ring == this.numberOfRings) {
+            return this;
+        }
+
+        //If the ring is the internal rings.
+        if (ring != 1) {
+            resultGrid = upsideDownFlipInternalMatrix(this.squarelotron);
+        } else {
+            //complete Upside Down Flip.
+            resultGrid = upsideDownFlipCompleteMatrix(this.squarelotron);
+
+            // Restore values of the internal rings if there are.
+            for (int i = 1; i < size - 1; i++) {
+                for (int j = 1; j < size - 1; j++) {
+                    resultGrid[i][j] = this.squarelotron[i][j];
+                }
+            }
+        }
+        return new Squarelotron(resultGrid);
+    }
+
+    /**
+     * flipExternalRing.
+     *
+     * @param matrix to be flip.
+     * @return the matrix result.
+     */
+    private int[][] flipExternalRing(final int[][] matrix) {
+        int[][] resultGrid = duplicateGrid(matrix);
+        int size = matrix.length;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "size" which hides the field declared at line 10. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)